src/org/opensolaris/opengrok/web/JSONSearchServlet.java

```diff
@@ -109,6 +111,13 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             result.put(PARAM_HIST, hist);
         }
 
+        if (type != null) {
+            type = URLDecoder.decode(type);
+            engine.setType(type);
+            valid = true;
+            result.put(PARAM_TYPE, type);
+        }
+        
```

Comment:
how is this relevant?

Reply:
oh I see the issue

Reply:
yep, just modified it while I was there.